src/bindings/python/src/openvino/frontend/pytorch/inlined_extension.py

```diff
@@ -0,0 +1,260 @@
+    elif isinstance(packed, list):
+        packer_result = []
+        for el in packed:
+            unpacked, packer, index = unpack(el, types, index)
+            unpacked_result += unpacked
+            packer_result.append(packer)
+    elif isinstance(packed, dict):
+        packer_result = {}
+        for key, el in packed.items():
+            unpacked, packer, index = unpack(el, types, index)
+            unpacked_result += unpacked
+            packer_result[key] = packer
+    elif isinstance(packed, types):
+        unpacked_result = (packed,)
+        packer_result = index
+        index += 1
+    else:
+        packer_result = ConstWrap(packed)
+    return unpacked_result, packer_result, index
+
+
+def pack(unpacked: tuple, packer: Any) -> Any:
+    """Packs unpacked elements back into their original structure."""
+    if isinstance(packer, (tuple, list)):
+        return type(packer)(pack(unpacked, el) for el in packer)
+    if isinstance(packer, dict):
+        return {k: pack(unpacked, v) for k, v in packer.items()}
+    if isinstance(packer, ConstWrap):
+        return packer.value
+    return unpacked[packer]
+
+
+global_counter_id = 0
+
+
+def make_custom_op_class(func: Callable,
+                         input_signature: tuple[tuple[Any, Any], ...],
+                         output_signature: tuple[tuple[Any, Any], ...],
+                         input_packer: Any,
+                         output_packer: Any) -> type:
+    """Creates a custom operation class from a function and signatures."""
+    global global_counter_id
+
+    class InlinedCustomOp(ov.Op):  # type: ignore
+        class_type_info = ov.runtime.DiscreteTypeInfo(
+            "InlinedCustomOp", "extension")
+
+        def __init__(self, *args: Any) -> None:
+            # TODO: What about attributes?
+            super().__init__(self, args)
+            # `id` attribute distinguishes different instances of the same
+            # class, we need it because different instances may have different
+            # behavior
+            self.attrs = {"id": global_counter_id}
+            self.constructor_validate_and_infer_types()
+
+        def evaluate(self, outputs: Any, inputs: Any) -> bool:
+            # TODO: Check memory sharing
+            inputs_torch = tuple(torch.from_numpy(input.data)
+                                 for input in inputs)
+            args, kwargs = pack(inputs_torch, input_packer)
+            result = func(*args, **kwargs)
+            result, result_packer, _ = unpack(result, torch.Tensor)
+            assert result_packer == output_packer
+            for i, tensor in enumerate(result):
+                if isinstance(tensor, torch.Tensor):
+                    ov_t = ov.Tensor(tensor.numpy(
+                        force=True), shared_memory=True)
+                else:
+                    ov_t = ov.Tensor(np.array(tensor), shared_memory=True)
+                # TODO: set the output tensor directly without copying
```

Comment:
minor: Do we plan to address the multiple TODO-s in this file? should we create a ticket and add a ticket number to these comments